observability/tracing.go

```diff
@@ -171,29 +173,76 @@ func getExporter(collectorURL string) (sdktrace.SpanExporter, error) {
 					)
 				} else {
 					log.Debugf("Creating OpenTelemetry collector with URL https://%s", collectorURL)
-
-					client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
-						tracingOptions,
-					)
+					// That's mainly for testing
+					if tracingOpt.Otel.SkipVerify {
+						log.Trace("OpenTelemetry collector will not verify the remote certificate")
+						client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
+							otlptracehttp.WithTLSClientConfig(&tls.Config{InsecureSkipVerify: true}),
+							tracingOptions,
+						)
+					} else {
+						client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
+							tracingOptions,
+						)
+					}
 				}
 
 				ctx := context.Background()
-				exporter, err := otlptrace.New(ctx, client)
-				return exporter, err
+				httpExporter, err2 := otlptrace.New(ctx, client)
+				return httpExporter, err2
 			} else {
 				if tracingOpt.Otel.Protocol == GRPC {
 					log.Debugf("Creating OpenTelemetry grpc collector with URL %s", collectorURL)
 					ctx := context.Background()
-					ctx, cancel := context.WithTimeout(ctx, time.Second)
+					ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 					defer cancel()
 
-					// TODO: Support TLS
-					exporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure(),
-						otlptracegrpc.WithEndpoint(collectorURL),
-						otlptracegrpc.WithDialOption(grpc.WithBlock()),
-					)
+					var grpcExporter *otlptrace.Exporter
+
+					if tracingOpt.Otel.TlsEnabled {
+						var creds credentials.TransportCredentials
+						// That's mainly for testing
+						if tracingOpt.Otel.SkipVerify {
+							log.Trace("OpenTelemetry collector will not verify the remote certificate")
+							tlsConfig := &tls.Config{
+								InsecureSkipVerify: true,
+							}
+							creds = credentials.NewTLS(tlsConfig)
+						} else {
+							cert_name := tracingOpt.Otel.CaName
+							if cert_name != "" {
+								var err2 error
+								creds, err2 = credentials.NewClientTLSFromFile(cert_name, "")
+								if err2 != nil {
+									log.Fatalf("Error loading certificate: %v", err2)
+									return nil, err2
+								}
+							} else {
+								log.Fatalf("ca_name is required")
+								return nil, fmt.Errorf("ca_name is required")
+							}
+						}
```

Comment:
Again I think you can do away with a lot of the nesting by simply checking the value and returning early rather than having an `if/else` branch. e.g.
```
if cert_name == "" {
    return nil, fmt.Errorf("ca_name is required")
 }

creds, err = credentials.NewClientTLSFromFile(cert_name, "")
if err != nil {
	return nil, err
}

```

Reply:
Updated